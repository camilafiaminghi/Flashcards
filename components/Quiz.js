import React, { Component } from 'react'
import { NavigationActions } from 'react-navigation'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { textColorInverse, textColor, pColor, pColorLight, pColorDark, sColor, sColorLight, sColorDark } from '../utils/colors'
import Question from './../components/Question'
import Answer from './../components/Answer'
import QuizScores from './../components/QuizScores'

export default class Quiz extends Component {

	static navigationOptions = ({ navigation }) => {
		return {
			title: navigation.getParam('current', 'Quiz')
		}
	}

	state = {
		current: 0,
		scored: 0,
		showAnswer: false
	}

	handleNext = () => {
		const { navigation } = this.props
		const { current } = this.state
		const { deck } = navigation.state.params

		this.setState((state) => ({
			...state,
			current: state.current + 1,
			showAnswer: false
		}))

		if ( current + 1 === deck.questions.length ) {
			navigation.setParams({current: `${deck.title}`})
		} else {
			navigation.setParams({current: `${deck.title} ${current + 2}/${deck.questions.length}`})
		}
	}

	handleScore = () => {
		const { scored } = this.state

		this.setState((state) => ({
			...state,
			scored: scored + 1
		}))

		this.handleNext()
	}

	toggleSide = () => {
		const { showAnswer } = this.state

		this.setState((state) => ({
			...state,
			showAnswer: !showAnswer
		}))
	}

	componentDidMount() {
		const { current } = this.state
		const { navigation } = this.props
		const { deck } = navigation.state.params

		navigation.setParams({current: `${deck.title} ${current + 1}/${deck.questions.length}`})
	}

	render() {
		const { current, showAnswer, scored } = this.state
		const { deck } = this.props.navigation.state.params

		if ( current === deck.questions.length ) {
			return (
				<QuizScores
					scored={scored}
					len={deck.questions.length} />
			)
		}

		const { question, answer } = deck.questions[current]

		return (
			<View style={styles.container}>
				<View style={styles.card}>
					{ (!showAnswer) &&
						<Question
							question={question}
							toggleSide={this.toggleSide} />
					}

					{ (showAnswer) &&
						<Answer
							answer={answer}
							handleScore={this.handleScore}
							handleNext={this.handleNext} />
					}
				</View>
			</View>
		)
	}
}

/* STYLES */
styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: sColorLight,
    paddingTop: 4,
    paddingBottom: 4
  },
  card: {
  	flex: 1,
  	flexGrow: 1,
  	alignItems: 'center',
  	justifyContent: 'center',
  	backgroundColor: 'white',
  	margin: 8,
		padding: 8,
  	borderRadius: 4,
  },
  cardTitle: {
  	marginBottom: 40,
  }
})
