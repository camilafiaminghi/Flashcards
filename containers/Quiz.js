import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { NavigationActions } from 'react-navigation'
import { connect } from 'react-redux'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { gray, lightPurp, purple, white } from '../utils/colors'
import Question from './../components/Question'
import Answer from './../components/Answer'
import QuizScores from './../components/QuizScores'

export class Quiz extends Component {

	static navigationOptions = ({ navigation }) => {
		return {
			title: navigation.getParam('current', 'Quiz')
		}
	}

	static propTypes = {
		entryId: PropTypes.string.isRequired,
		questions: PropTypes.array.isRequired
	}

	state = {
		current: 0,
		scores: 0,
		showAnswer: false
	}

	handleNext = () => {
		const { entryId, questions, navigation } = this.props
		const { current } = this.state

		this.setState((state) => ({
			...state,
			current: state.current + 1,
			showAnswer: false
		}))

		if ( current + 1 === questions.length ) {
			navigation.setParams({current: `${entryId} Scores`})
		} else {
			navigation.setParams({current: `${entryId} ${current + 2}/${questions.length}`})
		}
	}

	handleScore = () => {

		/* increment score if correct is triggered */
		/* change to next question */
		/* if not has next question show score in percents */
	}

	toggleSide = () => {
		const { showAnswer } = this.state

		this.setState((state) => ({
			...state,
			showAnswer: !showAnswer
		}))
	}

	componentDidMount() {
		const { current, scores } = this.state
		const { entryId, questions, navigation } = this.props
		navigation.setParams({current: `${entryId} ${current + 1}/${questions.length}`})
	}

	render() {
		const { questions } = this.props
		const { current, showAnswer, scores } = this.state

		if ( current === questions.length ) {
			return (<QuizScores scores={scores} />)
		}

		const { question, answer } = questions[current]

		return (
			<View style={styles.container}>
				<View style={styles.card}>
					{ (!showAnswer) &&
						<Question
							question={question}
							toggleSide={this.toggleSide} />
					}

					{ (showAnswer) &&
						<Answer
							answer={answer}
							handleScore={this.handleScore}
							handleNext={this.handleNext} />
					}
				</View>
			</View>
		)
	}
}

export const mapStateToProps = ({ entries }, { navigation }) => {
	const { entryId } = navigation.state.params

	return {
		entryId,
		questions: entries[entryId].questions
	}
}

export default connect(mapStateToProps)(Quiz)

/* STYLES */
export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f2f2f2'
  },
  card: {
  	flex: 1,
  	flexGrow: 1,
  	alignItems: 'center',
  	justifyContent: 'center',
  	backgroundColor: 'white',
  	margin: 8,
		padding: 8,
  	borderRadius: 4,
  },
  cardTitle: {
  	marginBottom: 40,
  }
})
