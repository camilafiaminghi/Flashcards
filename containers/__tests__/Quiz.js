import React, { Component } from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Quiz } from '../Quiz'
import mockStore from '../../__helpers__/entries'

let wrapper
const configStore = configureMockStore([thunk])
const store = configStore(mockStore)
const navigation = {
	navigate: jest.fn(),
	state: {
		params: {
			entryId: Object.keys(mockStore)[0]
		}
	},
	getParam: jest.fn(),
	setParams: jest.fn()
}
const props = {
	navigation,
	questions: mockStore[Object.keys(mockStore)[0]].questions
}

describe('<Quiz />', () => {

	beforeEach(() => {
		wrapper = renderer.create(<Provider store={store}><Quiz {...props} /></Provider>)

		console.log(wrapper.toJSON())
	})

	it('should be present', () => {
		// let component = renderer.create(<Quiz/>).toJSON()
		// console.log(component)
	})

})
