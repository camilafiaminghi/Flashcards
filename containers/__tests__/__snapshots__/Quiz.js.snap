// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quiz /> matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efefef",
      "flex": 1,
      "paddingBottom": 4,
      "paddingTop": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 4,
        "flex": 1,
        "flexGrow": 1,
        "justifyContent": "center",
        "margin": 8,
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontSize": 28,
            "marginBottom": 20,
            "marginLeft": 10,
            "marginRight": 10,
            "textAlign": "center",
          }
        }
      >
        What is React?
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 4,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6f79a8",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          SHOW ANSWER
        </Text>
      </View>
    </View>
  </View>
</View>
`;
